Source/WebCore/page/DOMSelection.cpp

```diff
@@ -531,7 +553,14 @@ RefPtr<Node> DOMSelection::shadowAdjustedNode(const Position& position) const
 
     if (frame()->settings().liveRangeSelectionEnabled()) {
         auto node = position.containerNode();
-        return !node || node->isInShadowTree() ? nullptr : node;
+        if (!node)
+            return nullptr;
+        if (node->isInShadowTree()) {
```

Comment:
This isn't quite right. We need to find the outermost shadow host instead.
frame()->document()->ancestorNodeInThisScope is the right thing to use here as the non-live range version does.

Reply:
Good catch — as you pointed out over Slack, we can just remove this `liveRangeSelectionEnabled()` check and use the shared codepath below.